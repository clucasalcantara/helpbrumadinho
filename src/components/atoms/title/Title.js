import React from 'react'
import PropTypes from 'prop-types'
import { Header } from 'semantic-ui-react'

const getSize = size => {
  switch (size) {
    case 'small':
      return 'h4'
    case 'medium':
      return 'h3'
    case 'big':
      return 'h1'

    default:
      return 'h2'
  }
}
const Title = ({ content, size }) => (
  <Header as={getSize(size)}>{content}</Header>
)

Title.propTypes = {
  content: PropTypes.string.isRequired,
  size: PropTypes.string
}

Title.defaultProps = {
  size: null
}

export default Title
