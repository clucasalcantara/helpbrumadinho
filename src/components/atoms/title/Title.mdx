---
name: Title
menu: Atoms
---

import { Playground, PropsTable } from 'docz'
import Title from './Title'

# Title component

## Basic usage

<Playground>
  <Title content={'Work with Coffee'} />
</Playground>

## Properties

<PropsTable of={Title} />
