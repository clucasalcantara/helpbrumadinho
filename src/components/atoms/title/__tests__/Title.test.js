import React from 'react'
import { shallow } from 'enzyme'
import Title from '../Title'

describe('Title component', () => {
  it('should renders without crash', () => {
    const wrapper = shallow(<Title content="Work with coffee" />)

    expect(wrapper).toMatchSnapshot()
  })
})
