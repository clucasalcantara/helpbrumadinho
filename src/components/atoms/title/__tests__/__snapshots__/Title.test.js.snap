// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Title component should renders without crash 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Title
    content="Work with coffee"
    size={null}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "as": "h2",
      "children": "Work with coffee",
    },
    "ref": null,
    "rendered": "Work with coffee",
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "as": "h2",
        "children": "Work with coffee",
      },
      "ref": null,
      "rendered": "Work with coffee",
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
